zebra-consensus/src/transaction/check.rs

```diff
@@ -124,10 +125,8 @@ pub fn has_enough_orchard_flags(tx: &Transaction) -> Result<(), TransactionError
 ///
 /// <https://zips.z.cash/protocol/protocol.pdf#txnconsensus>
 pub fn coinbase_tx_no_prevout_joinsplit_spend(tx: &Transaction) -> Result<(), TransactionError> {
-    if tx.has_valid_coinbase_transaction_inputs() {
-        if tx.contains_prevout_input() {
-            return Err(TransactionError::CoinbaseHasPrevOutInput);
```

Comment:
As pointed out in the Change History in the spec, this check is redundant per the definition of "coinbase transaction".

Reply:
Question: Do we want to `assert!(!tx.contains_prevout_input());` here?

Reply:
I think it doesn't add much value since the `is_coinbase()` check is right before and it ensures that this is true.

I also realized that `contains_prevout_input()` is no longer being used, so that's another small reason to not do the assert. (I went ahead and removed the method it in 0ec750fdb)

Reply:
`TransactionError::CoinbaseHasPrevOutInput` is now unused, can you remove it as well?

Reply:
Done in b6ce777e2